src/Architecture.cpp

```diff
@@ -302,6 +302,10 @@ void Architecture::minimumNumberOfSwaps(std::vector<unsigned short>& permutation
         qubits.insert(q);
     }
 
+    if (qubits.size() != permutation.size()) {
+        throw std::runtime_error("Architecture::minimumNumberOfSwaps: permutation contains duplicates");
+    }
```

Comment:
this change is not really related to this PR...

Reply:
It provides a more helpful error message when something with the encodings goes wrong, either the naive ones or the ones provided by logicblocks. It is thus tangentially related.